### PR TITLE
Implement health status endpoint

### LINKED ISSUE #143: Implement endpoint for health check
Bugs in dependency libraries can cause the karapace APIs to become unresponsive, restarting the process is usually a valid and quick fix for the problem. To properly implement the restart a `/status` endpoint is necessary, this endpoint has to peform some request through the kafka client to make sure communication is working properly.

### PR DESCRIPTION
**Pull request by tvainika**
# About this change - What it does

Add health status endpoint
    
Fixes #143

# Why this way

Implement health status endpoint which returns status of schema registry.  This by design avoids long running operations, and queries status from internal state variables acquiring only minimal locks.  This is designed to robustly and quickly return internal status of Karapace.

### CODE REVIEW

File: karapace/master_coordinator.py [hunk 1]
```diff
@@ -212,6 +222,16 @@ def init_schema_coordinator(self) -> None:
         )
         self.schema_coordinator_ready.set()
 
+    def get_coordinator_status(self) -> SchemaCoordinatorStatus:
+        generation = self.sc.generation() if self.sc is not None else None
+        return SchemaCoordinatorStatus(
+            is_primary=self.sc.are_we_master if self.sc is not None else None,
+            is_primary_eligible=cast(bool, self.config["master_eligibility"]),
```
Comment (jjaakola-aiven) [inline karapace/master_coordinator.py:229]: This is available from `SchemaCoordinator.master_eligibility`.
https://github.com/aiven/karapace/blob/2ab4e832549274442ac13a175c821a5e9f27890b/karapace/master_coordinator.py#L73

File: karapace/master_coordinator.py [hunk 2]
```diff
@@ -212,6 +222,16 @@ def init_schema_coordinator(self) -> None:
...
+            primary_url=self.sc.master_url if self.sc is not None else None,
+            is_running=self.schema_coordinator_ready.is_set(),
```
Comment (jjaakola-aiven) [inline karapace/master_coordinator.py:231]: The ready flag been set is known but not if the thread is actually still running, hanging or failing. The more correct would be `self.running`, but it does not also tell if the loop is performing.

File: karapace/schema_registry_apis.py [hunk 3]
```diff
@@ -84,6 +84,22 @@ def __init__(self, config: Config) -> None:
...
+        resp["schema_registry_ready"] = self.schema_registry.schema_reader.ready
+        resp["schema_registry_reader_offset"] = self.schema_registry.schema_reader.offset
+        resp["schema_registry_reader_highest_offset"] = self.schema_registry.schema_reader.highest_offset
+        cs = self.schema_registry.mc.get_coordinator_status()
+        resp["schema_registry_is_primary"] = cs.is_primary
+        resp["schema_registry_is_primary_eligible"] = cs.is_primary_eligible
+        resp["schema_registry_primary_url"] = cs.primary_url
+        resp["schema_registry_coordinator_running"] = cs.is_running
+        resp["schema_registry_coordinator_generation_id"] = cs.group_generation_id
+        return resp
```
Comment (jjaakola-aiven) [inline karapace/schema_registry_apis.py:102]: The highest offset is only read on startup so I'd make that explicit on the field name in JSON. See also other field name suggestions in the proposed format below.
And I would add also the current offset. And maybe add the highest offset from `OffsetWatcher`, the difference would be the lag and it should be small (0 or 1 in most cases). If the lag goes up then there is a problem.

Maybe format could be this?
```
{
  "schema_registry": {
    "authfile_modified_ts": self._auth.auth...,
    "ready": ..,
    "highest_offset_on_start": ..,
    "current_offset": ...,
    ....
  }
}
```

File: karapace/master_coordinator.py (hunk 1 above)
Comment (tvainika) [reply karapace/master_coordinator.py:229]: However `SchemaCoordinator` might not be initialized in the beginning ie. `self.sc is None`...

File: karapace/master_coordinator.py (hunk 2 above)
Comment (tvainika) [reply karapace/master_coordinator.py:231]: Good point. Looking more, I'm thinking if `Thread.is_alive()` would serve best way here...

File: karapace/schema_registry_apis.py (hunk 3 above)
Comment (tvainika) [reply karapace/schema_registry_apis.py:102]: I considered level structured response as there can be REST proxy related state variables too, but ended up with flat response for sake of simplicity.

Current code exposes current offset ie. the offset up to where schema reader has been reading messages. Did you meant to suggest about naming `schema_registry_reader_offset` -> `...current_offset` or did I miss something?

Incorporating `OffsetsWatcher`'s offset it is waiting into same exposed highest offset sounds like a plan to me.

**Review by jjaakola-aiven — APPROVED** [review]

**Pull request merged**